lib/statsd.js

```diff
@@ -62,6 +62,15 @@ function makeStatsD(options, logger) {
         statsd = new StatsD(statsdOptions);
     }
 
+    // Support creating sub-metric clients with a fixed prefix. This is useful
+    // for systematically categorizing metrics per-request, by using a
+    // specific logger.
+    statsd.makeChild = function(name) {
+        var childOptions = Object.assign({}, options);
+        childOptions._prefix = statsdOptions.prefix + normalizeName(name) + '.';
```

Comment:
options._prefix is a private attribute (as indicated by `_`) that we construct from normalized parts.

Regarding the dot, there was indeed an issue in that we ended up with two trailing dots for child metrics clients. Removed the trailing dot from the _prefix construction to avoid this. Good catch!